src/TConsole.cpp

```diff
@@ -1840,9 +1849,24 @@ void TConsole::slot_searchBufferUp()
         } while (searchX > -1);
 
         if (found) {
+            mpHost->setCaretEnabled(true);
+            mUpperPane->initializeCaret();
+            moveCursor(searchX, searchY);
+            mUpperPane->setCaretPosition(searchY, searchX);
+            mUpperPane->updateCaret();
+            mUpperPane->setFocusPolicy(Qt::StrongFocus);
+            mUpperPane->setFocusProxy(nullptr);
+            mUpperPane->setFocus();
+            
+            // Scroll to show the match
             scrollUp(buffer.mCursorY - searchY - 3);
             mUpperPane->forceUpdate();
             mCurrentSearchResult = searchY;
+            
+            // Announce match to screen readers
+            if (auto* announcer = findChild<Announcer*>()) {
+                announcer->announce(buffer.lineBuffer[searchY]);
```

Comment:
I am not sure this is needed given what @ironcross32 is saying. Is Voiceover already not speaking things from the setCaretPosition changes?

Reply:
I had added these to read the whole line, let me see, also, I have not pushed the caret update change yet. will do.

Reply:
@vadi2 this will be needed, and I will move them to the place where highlighting is being done, that will be helpful, although, I am observing that voiceover starts dictating words in the active window and output window content is not read sometimes